src/integrationTest/java/com/mongodb/spark/sql/connector/read/partitioner/ShardedPartitionerTest.java

```diff
@@ -0,0 +1,219 @@
+    // Single host
+    assertIterableEquals(
+        singletonList("sh0.example.com"), PARTITIONER.getHosts("tic/sh0.example.com:27018"));
+    // Multiple hosts
+    assertIterableEquals(
+        asList("sh0.rs1.example.com", "sh0.rs2.example.com", "sh0.rs3.example.com"),
+        PARTITIONER.getHosts(
+            "tic/sh0.rs1.example.com:27018,sh0.rs2.example.com:27018,sh0.rs3.example.com:27018"));
+  }
+
+  private void assertPartitioner(final ReadConfig readConfig) {
+    List<MongoInputPartition> mongoInputPartitions = PARTITIONER.generatePartitions(readConfig);
+    assertPartitionsAreOrderedAndBounded(
+        mongoInputPartitions, PARTITIONER.getShardKeys(readConfig));
+
+    List<BsonDocument> aggregationPipeline = readConfig.getAggregationPipeline();
+    if (!aggregationPipeline.isEmpty()) {
+      mongoInputPartitions.forEach(
+          mongoInputPartition ->
+              assertTrue(mongoInputPartition.getPipeline().containsAll(aggregationPipeline)));
+    }
+
+    assertPartitionerCoversAllData(PARTITIONER, readConfig);
+  }
+
+  /**
+   * Different versions of MongoDB will shard chunks slightly differently.
+   *
+   * <p>This prevents explicitly comparing the shard chunk partition ranges against expected ranges.
+   * Instead, the MongoInputPartition ranges are checked to ensure they are bounded by continuous
+   * `$lt` and `$gte` ranges.
+   */
+  private void assertPartitionsAreOrderedAndBounded(
+      final List<MongoInputPartition> mongoInputPartitions, final List<String> shardKeys) {
+    assertFalse(mongoInputPartitions.isEmpty(), "No partitions were produced");
+    assertMongoPartitionHasBounds(getMatchStage(mongoInputPartitions.get(0)), shardKeys, "$lt");
+
+    for (int i = 1; i < mongoInputPartitions.size() - 1; i++) {
+      BsonDocument ltMatch = getMatchStage(mongoInputPartitions.get(i));
+      BsonDocument gteMatch = getMatchStage(mongoInputPartitions.get(i + 1));
+      assertMongoPartitionBounds(ltMatch, gteMatch, shardKeys);
+    }
+
+    if (mongoInputPartitions.size() > 1) {
+      assertMongoPartitionHasBounds(
+          getMatchStage(mongoInputPartitions.get(mongoInputPartitions.size() - 1)),
+          shardKeys,
+          "$gte");
+    }
+  }
+
+  private void assertPartitionsAreOrderedAndBounded(final ReadConfig readConfig) {
+    assertPartitionsAreOrderedAndBounded(
+        PARTITIONER.generatePartitions(readConfig), PARTITIONER.getShardKeys(readConfig));
+  }
+
+  private BsonDocument getMatchStage(final MongoInputPartition mongoInputPartition) {
+    return mongoInputPartition.getPipeline().isEmpty()
+        ? new BsonDocument()
+        : mongoInputPartition.getPipeline().get(0).getDocument("$match", new BsonDocument());
+  }
+
+  private void assertMongoPartitionHasBounds(
+      final BsonDocument matchStage,
+      final List<String> shardKeys,
+      final String queryComparisonOperator) {
+    shardKeys.forEach(
+        shardKey ->
+            assumeTrue(
```

Comment:
There appears to be a typo here: this needs to be `assertTrue`.

Reply:
Ack